yaml/_loader_state.ts

```diff
@@ -529,15 +534,15 @@ export class LoaderState {
     let lineBreaks = 0;
     let ch = this.peek();
 
-    while (ch !== 0) {
+    while (!this.#isEof()) {
```

Comment:
So now we're getting `this.peek()` twice. This appears to be the case for the other changed lines also.